lib/data/file/backend.js

```diff
@@ -65,11 +74,15 @@ export const backend = {
                 return callback(errors.InternalError);
             })
             .on('end', () => {
-                log.debug('finished writing data', { key });
-                fs.fsync(fd, () => {
-                    fs.close(fd);
-                    return callback(null, key);
-                });
+                streamEnd = true;
+                // Only in flowing mode when fs.write finished
+                if (request._readableState.flowing) {
```

Comment:
@ThibaultRiviere, isn't the finish event for a writable stream and the request here is acting as a readable? @RachedBenMustapha, pointed out that with my suggestion if we have more than one chunk being finished up, we could still end early.  He suggests that we push each chunk into an array on each data event and push null into the array when end is called on the readable.  Then in the on.("data") block, we pop off of the end of the array and write it if it is a chunk or close the file and call the callback if the end of the array contains null.